app-backend/db/src/test/scala/com/azavea/rf/database/OrganizationDaoSpec.scala

```diff
@@ -1,30 +1,85 @@
 package com.azavea.rf.database
 
 import com.azavea.rf.datamodel.Organization
+import com.azavea.rf.datamodel.Generators.Implicits._
 import com.azavea.rf.database.Implicits._
 
 import doobie._, doobie.implicits._
 import cats._, cats.data._, cats.effect.IO
 import cats.syntax.either._
 import doobie.postgres._, doobie.postgres.implicits._
 import doobie.scalatest.imports._
+import org.scalacheck.Prop.forAll
 import org.scalatest._
+import org.scalatest.prop.Checkers
 
 
-class OrganizationDaoSpec extends FunSuite with Matchers with IOChecker with DBTestConfig {
+class OrganizationDaoSpec extends FunSuite with Matchers with Checkers with DBTestConfig {
 
-  test("insertion") {
-    val testName = "test org"
+  // create
+  test("insert an organization from a name") {
+    check {
+      forAll(
+        (name: String) => {
+          val withoutNull = name.filter( _ != '\u0000' ).mkString
+          OrganizationDao.create(withoutNull).transact(xa).unsafeRunSync.name == withoutNull
+        }
+      )
+    }
+  }
 
-    val transaction = for {
-      orgIn <- OrganizationDao.create(testName)
-      orgOut <- OrganizationDao.query.filter(fr"id = ${orgIn.id}").selectQ.unique
-    } yield orgOut
+  // createOrganization
+  test("insert an organization from an Organization.Create") {
+    check {
+      forAll(
+        (orgCreate: Organization.Create) => {
+          OrganizationDao.createOrganization(orgCreate).transact(xa).unsafeRunSync.name == orgCreate.name
+        }
+      )
+    }
+  }
 
-    val result = transaction.transact(xa).unsafeRunSync
-    result.name shouldBe testName
+  // getOrganizationById
```

Comment:
If we're doing this, I feel like we should have a `getOrganizationList` method in `OrganizationDao`, with a corresponding test.
Is each test conducted as a separate transaction, or is it the entire set of tests?
If tests are conducted in the order that they are defined and they are all contained in a single transaction, then we don't need to run createOrganization to test getOrganizationById, and instead test both the list / getById functions together (list gets some orgs, then id re-fetches ones that were returned in the list or something)

Reply:
Ah yeah I wasn't sure whether to duplicate this information in each PR and I didn't include a reference to it, sorry.

In the anchor PR:

> getting values in a single transaction is necessary, because the test transactor has rollback turned on. If you foo.transact(xa).unsafeRunSync then bar.transact(xa).unsafeRunSync separately, the result of foo will have been rolled back by the time you bar. This requires a little bit of type gymnastics to make sure that you're testing against a state of the database that you understand.

Anyway yeah, we should test the `list` method also. I have that on the `BandDao` tests that I'm working on but didn't have it here. I'll add.